### PR TITLE
{docs} new help and parameter content guidelines

### PR DESCRIPTION
**Pull request by dbradish-microsoft**
The intent of this new doc is to provide guidance on reference command and parameter help.  The focus of this doc is three-fold
- in-line help
- (resulting in) autogenerated reference content
- Minimum conceptual content required to publish a new reference group or subgroup

The guideline for an ```output code block is under discussion.  Ideally, this should be in autogen content, and autogenerated from source code (vs hard-coded in _help.py).

Please put all comments in `Files changed` -->` Review`

### CODE REVIEW

**Comment by yonzhan** [comment]
all service

**Comment by dbradish-microsoft** [comment]
@yonzhan / @jsntcy, Might it be possible to have this doc reviewed and published before the end of August, 2023?

**Comment by jsntcy** [comment]
> @yonzhan / @jsntcy, Might it be possible to have this doc reviewed and published before the end of August, 2023?

Need more team members to review.
@evelyn-ys, @kairu-ms to help review.

**Comment by yonzhan** [comment]
This PR needs more time for CLI team members to review and it can be merged by 09/26/2023.

File: doc/reference_doc_guidelines.md [hunk 1]
```diff
@@ -0,0 +1,295 @@
...
+
+  For more information on syntax differences between Bash and PowerShell, see these articles:
+    - [Quoting issues with PowerShell](https://github.com/Azure/azure-cli/blob/dev/doc/quoting-issues-with-powershell.md)
+    - [Use quotation marks in Azure CLI parameters](./use-cli-effectively.md#use-quotation-marks-in-parameters)
+    - Compare syntax of CMD, PowerShell and Bash in [Query command output using JMESPath](./query-azure-cli.md)
+    - [Error handling for the Azure CLI in PowerShell](./use-cli-effectively.md#error-handling-for-azure-cli-in-powershell)
+
+    **Do not replicate code blocks for line break differences.**  The new Azure CLI code block <kbd>Copy</kbd> button removes Bash and PowerShell line break characters. Only replicate code blocks for syntax differences.
+
+- Provide **\```output** code blocks. When developing scripts, customers will likely combine two or more create, read, update and delete operations. With no documentation about the format of the returned object, customers have no choice but to run each command one by one and analyze the response.
```
Comment (jiasli) [inline doc/reference_doc_guidelines.md:26]: Providing documentation for output JSON can be challenging. We previously had this conversation in https://github.com/Azure/azure-cli/issues/17332.

Comment (dbradish-microsoft) [reply doc/reference_doc_guidelines.md:26]: @jiasli, understood, but this continues to be a request by @dcaro.  Let's compromise and I'll move this to the conceptual requirements section until there is both a **plan** and a **place** for this in the source code.

**Comment by dbradish-microsoft** [comment]
@jiasli, @jsntcy , @yonzhan , I've responded to @jiasli's comment and this PR is ready for review again.

File: doc/reference_doc_guidelines.md [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+> [!div class="checklist"]
+>
+> - Examples have been tested in both Bash and PowerShell environments
+> - At least two examples have been given for each command.
+> - Bash line continuation characters have been used for improved readability.
+> - HTML tags have not been used in examples.
+
+### Parameter checklist
+
+> [!div class="checklist"]
```
Comment (jsntcy) [inline doc/reference_doc_guidelines.md:294]: I'm curious about the purpose that we add "[!div class="checklist"]".

File: doc/reference_doc_guidelines.md [hunk 3]
```diff
@@ -0,0 +1,299 @@
...
+  - Compare syntax of CMD, PowerShell and Bash in [Query command output using JMESPath](./query-azure-cli.md)
+  - [Error handling for the Azure CLI in PowerShell](./use-cli-effectively.md#error-handling-for-azure-cli-in-powershell)
+
+    **Do not replicate code blocks for line break differences.** The new Azure CLI code block <kbd>Copy</kbd> button removes Bash and PowerShell line break characters. Only replicate code blocks for syntax differences.
+
+## Conceptual content
+
+### Minimum content
+
+Minimum conceptual content for the Azure CLI is one conceptual article for every command group and subgroup that creates or adds a billable Azure resource. For example:
```
Comment (jsntcy) [inline doc/reference_doc_guidelines.md:30]: Which team will add/update conceptual articles? I assume your team will help on that, is it true?

**Comment by jsntcy** [comment]
> @jiasli, @jsntcy , @yonzhan , I've responded to @jiasli's comment and this PR is ready for review again.

Overall LGTM, just with some comments.

**Review by jsntcy — APPROVED** [review]

**Pull request merged**